upstairs/src/volume.rs

```diff
@@ -369,6 +369,18 @@ impl BlockIO for Volume {
         Ok(())
     }
 
+    fn deactivate(&self) -> Result<BlockReqWaiter, CrucibleError> {
+        for sub_volume in &self.sub_volumes {
+            sub_volume.deactivate()?;
+        }
+
+        if let Some(ref read_only_parent) = &self.read_only_parent {
+            read_only_parent.deactivate()?;
+        }
+
+        BlockReqWaiter::immediate()
+    }
```

Comment:
on lines 374/378, we should probably block-wait on the BlockReqWaiters returned by the subvolume/parent deactivations before returning a BlockReqWaiter::immediate()

Reply:
Definitely! I totally missed this. Fixed in 0e27718de864ee59b6c1f17b65e0265737818424.

I also added a `#[must_use]` for `BlockReqWaiter` in 523d63e, which caught some other misses.

Reply:
Update: it was probably this that was causing the looping, I don't see it anymore.

Reply:
oh awesome, I know I definitely almost forgot to wait on a couple while I was writing crudd.